fetch/fetch.go

```diff
@@ -168,13 +183,67 @@ func (f *httpFetcher) newRequest() (req *http.Request, err error) {
 	return req, nil
 }
 
-/*
-Author:  Benjamin Bengfort
-Author:  Rebecca Bilbro
-Created: Mon Apr 29 06:43:36 2019 -0400
+//===========================================================================
+// HTML Fetcher
+//===========================================================================
 
-Copyright (C) 2019 Kansas Labs
-For license information, see LICENSE.txt
+// The htmlFetcher uses GET requests to retrieve the html containing the full text
+// of articles of feeds with a Baleen-specific http client.
+type htmlFetcher struct {
+	url string // the url of the article full text
+}
 
-ID: fetch.go [d6dba70] benjamin@bengfort.com $
-*/
+func (f *htmlFetcher) Fetch() (content string, err error) {
+	var req *http.Request
+	if req, err = f.newRequest(); err != nil {
+		return "", err
+	}
+
+	var rep *http.Response
+	if rep, err = client.Do(req); err != nil {
+		return "", err
+	}
+
+	// Close the body of the response reader when we're done.
+	if rep != nil {
+		defer func() {
+			ce := rep.Body.Close()
+			if ce != nil {
+				err = ce
+			}
+		}()
+	}
+
+	// Check the status code of the response; note that 304 means not modified, but we
+	// are still returning a 304 error to signal to the Subscription that nothing has
+	// changed and that the feed is nil.
+	if rep.StatusCode < 200 || rep.StatusCode >= 300 {
+		return "", HTTPError{
+			Status: rep.Status,
+			Code:   rep.StatusCode,
+		}
+	}
+
+	// Parse the html
+	html, err := ioutil.ReadAll(rep.Body)
+	if err != nil {
+		log.Printf("no text parsed from html retrieved from %s", f.url)
+	}
+	content = string(html)
```

Comment:
RE: the encoding tasks, perhaps we could leave the content as `[]byte` for as long as possible, and only encode it to string when absolutely necessary?